### PR TITLE
Add VIA keymap to duckyPad, update M4cs keymap for duckyPad

### PR DESCRIPTION
**Pull request by M4cs**
## Description

Adds VIA support for duckyPad macropads. Also updates my personal keymap to include my systeminfo layer. This also adds a python script to my keymap folder. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/protocol/usb_descriptor.h [hunk 1]
```diff
@@ -278,7 +278,7 @@ enum usb_endpoints {
 #define KEYBOARD_EPSIZE 8
 #define SHARED_EPSIZE 32
 #define MOUSE_EPSIZE 8
-#define RAW_EPSIZE 32
+#define RAW_EPSIZE 64
```
Comment (zvecr) [inline tmk_core/protocol/usb_descriptor.h:281]:
```suggestion
#define RAW_EPSIZE 32
```

File: keyboards/dekunukem/duckypad/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/keymaps/via/rules.mk:2]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: tmk_core/protocol/usb_descriptor.h (hunk 1 above)
Comment (M4cs) [reply tmk_core/protocol/usb_descriptor.h]: Whoops must have changed this when trying to test something, fixed 👍🏼

File: keyboards/dekunukem/duckypad/info.json [hunk 3]
```diff
@@ -1,8 +1,14 @@
 {
     "keyboard_name": "duckyPad",
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/info.json:2]: The changes to this file should be reverted, as most of this information is pulled from the config.h file.  And `keymap` isn't used at all.

File: keyboards/dekunukem/duckypad/bootloader_defs.h [hunk 4]
```diff
@@ -0,0 +1 @@
+#define STM32_BOOTLOADER_ADDRESS 0x1FFFC800
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/bootloader_defs.h:1]: This file is not needed.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dekunukem/duckypad/bootloader_defs.h (hunk 4 above)
Comment (M4cs) [reply keyboards/dekunukem/duckypad/bootloader_defs.h]: Fixed thanks

File: keyboards/dekunukem/duckypad/info.json (hunk 3 above)
Comment (M4cs) [reply keyboards/dekunukem/duckypad/info.json]: Reverted, thanks

File: keyboards/dekunukem/duckypad/config.h [hunk 5]
```diff
@@ -22,11 +22,12 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x0B91
-#define DEVICE_VER      0x0001
+#define VENDOR_ID       0x444E // "DN"
+#define PRODUCT_ID      0x4450 // "DP"
+#define DEVICE_VER      0x0001 // 1.0
 #define MANUFACTURER dekuNukem
 #define PRODUCT duckyPad
+#define DESCRIPTION duckyPad
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/config.h:30]:
```suggestion
```

File: keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c [hunk 6]
```diff
@@ -21,24 +21,30 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 
 enum my_keycodes {
-    Z0 = SAFE_RANGE, Z1, Z2,
-    Z3, Z4, Z5,
-    Z6, Z7, Z8,
-    Z9, Z10, Z11,
+    Z0 = SAFE_RANGE, Z1, Z2, \
+    Z3, Z4, Z5, \
+    Z6, Z7, Z8, \
+    Z9, Z10, Z11, \
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c:47]:
```suggestion
    Z0 = SAFE_RANGE, Z1, Z2,
    Z3, Z4, Z5,
    Z6, Z7, Z8,
    Z9, Z10, Z11,
```

File: keyboards/dekunukem/duckypad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#include <unistd.h>
+
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/via/keymap.c:24]:
```suggestion
```

File: keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c [hunk 8]
```diff
@@ -21,24 +21,30 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include QMK_KEYBOARD_H
 
 #include "stdio.h"
+#include "raw_hid.h"
 #include <unistd.h>
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c:25]:
```suggestion
```
I don't think this header is needed.

File: keyboards/dekunukem/duckypad/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  LAYOUT(
+      KC_KP_1, KC_KP_2, KC_KP_3,
+      KC_KP_4, KC_KP_5, KC_KP_6,
+      KC_KP_7, KC_KP_8, KC_KP_9,
+      KC_TRNS, KC_KP_0, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS
+  )
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/via/keymap.c:36]: You need three more empty layers here.

File: keyboards/dekunukem/duckypad/duckypad.h [hunk 10]
```diff
@@ -24,9 +24,9 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #define LAYOUT( \
     k00, k01, k02, \
-    k10, k11, k12, \
-    k20, k21, k22, \
-    k30, k31, k32, \
-    k40, k41, k42, \
-    kDown, kUp \
-) {{k00, k01, k02, k10, k11, k12, k20, k21, k22, k30, k31, k32, k40, k41, k42, kDown, kUp}}
+    k03, k04, k05, \
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/duckypad.h:27]: Why was this changed?

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c (hunk 8 above)
Comment (M4cs) [reply keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c:25]: Wouldn't this be needed since I'm using `raw_hid_receive` ?

File: keyboards/dekunukem/duckypad/duckypad.h (hunk 10 above)
Comment (M4cs) [reply keyboards/dekunukem/duckypad/duckypad.h:27]: My direct pins are not setup in a matrix so I felt this better represented the matrix defined in config.h

File: keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c (hunk 8 above)
Comment (fauxpark) [reply keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c:25]: I don't see where raw HID requires unistd.h.

Comment (M4cs) [reply keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c:25]: Oh my bad I read the diff incorrectly

**Review by drashna — APPROVED** [review]

**Comment by M4cs** [comment]
Is it possible to bump this up? I've made revisions for all requested changes

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

**Comment by M4cs** [comment]
> 
> 
> .

Fixed thanks

**Review by fauxpark — APPROVED** [review]

**Comment by M4cs** [comment]
@zvecr Would you be able to approve my changes?
